ddtrace/utils/install.py

```diff
@@ -0,0 +1,82 @@
+import sys
+
+from wrapt.importer import (
+    register_post_import_hook,
+    _post_import_hooks,
+    _post_import_hooks_lock,
+)
+from wrapt.decorators import synchronized
+
+
+def install_module_import_hook(modulename, modulehook):
+    """Installs a module import hook for a given module.
+
+    A flag is also stored and checked on the module to so that the module is
+    not instrumented more than once.
+    """
+    # wrap the module hook with an idempotence check
+    def check_patched_hook(module):
+        if module_patched(module):
+            return
+        _mark_module_patched(module)
+        modulehook(module)
+    setattr(check_patched_hook, '_datadog_hook', True)
+    register_post_import_hook(check_patched_hook, modulename)
+
+
+def _hook_matcher(hook):
+    """Matches functions that are hooks."""
+    return hasattr(hook, '_datadog_hook')
+
+
+def uninstall_module_import_hook(modulename):
+    """Uninstalls a module import hook for a module given its name.
+
+    Note: any patching done on the module will still apply, this function
+    simply removed/disables the import hook.
+    """
+    if modulename in sys.modules:
+        module = sys.modules[modulename]
+        _mark_module_unpatched(module)
```

Comment:
Not sure we want this?

What if we install, patch, then uninstall? The module should still be marked as patched?

Reply:
Yeah it's not ideal but `uninstall_module_import_hook` assumes that the caller is going to perform the unpatching logic and then call `uninstall_module_import_hook` and hence we mark it "unpatched". Previously I had exposed the `mark_module_unpatched` and then the convention would be for the integration to be in charge of marking it unpatched. Neither of these solutions seem particularly good.

Reply:
maybe `uninstall_module_import_hook_and_mark_unpatched` is a better more verbose solution?

Reply:
If we are moving away from "unpatching" do we need to mark it as unpatched at all?

We should have `patch()` -> setup hooks, `unpatch()` -> disable hooks  ?

Reply:
_Eventually_ yes I think. It's just tricky for what we do in the meantime.